src/components/modal/Modal.svelte

```diff
@@ -0,0 +1,408 @@
+                  <Button
+                    {...okButtonProps}
+                    type="{okType}"
+                    on:click="{onOkLocal}"
+                    loading="{confirmLoading}"
+                    focusOnMount="{autoFocusButton === 'ok'}">
+                    {okText}
+                  </Button>
+                </div>
+              </slot>
+            </div>
+          {/if}
+        </div>
+      </div>
+    </div>
+  </div>
+{/if}
+
+<script>
+  import { createEventDispatcher, onMount, onDestroy } from "svelte";
+  import { fade, fly } from "svelte/transition";
+  import {
+    CloseOutlined,
+    InfoCircleOutlined,
+    CheckCircleOutlined,
+    CloseCircleOutlined,
+    ExclamationCircleOutlined
+  } from "@/components/icons";
+  import Button from "../button/Button.svelte";
+  import { modalDestroyFunctions } from "./store";
+
+  const dispatch = createEventDispatcher();
+
+  // ********************** Props **********************
+
+  // If the modal is visible
+  export let visible = false;
+  // User has the option to remove the footer
+  export let footer = true;
+  // Modal width
+  export let width;
+  // Title for the header
+  export let title = "";
+  // If the ok button is in loading state
+  export let confirmLoading = false;
+  // Confirm and status modals need a differnet component sructure
+  export let modalType = "standard";
+  // Icon for confirm and status modals
+  export let icon = null;
+  // confirm and status modals can accept content as an option (string or component)
+  export let content = null;
+  // confirm and status modals need to call ok and cancel functions
+  export let onOk = null;
+  export let onCancel = null;
+  // Extra props for the buttons
+  export let okButtonProps = {};
+  export let cancelButtonProps = {};
+  // Type of button to use for OK (primary, danger, etc.)
+  export let okType = "primary";
+  // Text of the ok button
+  export let okText = "OK";
+  // Text of the cancel button
+  export let cancelText = "Cancel";
+  // Allow top and bottom positioning
+  export let verticalPosition = {};
+  // Center modal on page
+  export let centered = false;
+  // Visibility of the close button
+  export let closeable = true;
+  // Icon to use for the close button
+  export let closeIcon = CloseOutlined;
+  // Whether show mask or not
+  export let mask = true;
+  // Close the modal when the mask is clicked
+  export let maskClosable = true;
+  // Function to be called after the modal closes
+  export let afterClose = () => {};
+  // Functions for destroying the modal
+  export const destroy = () => {
+    localVisible = false;
+    scrollableBody = true;
+    afterClose();
+    // Remove the destroy function modalDestroyFunctions store
+    if (modalType !== "standard") {
+      modalDestroyFunctions.set(
+        $modalDestroyFunctions.filter(func => func !== destroy)
+      );
+    }
+  };
+  // Body style for modal body element
+  export let bodyStyle = "";
```

Comment:
For styles and classes I would suggest we use the approach I did with the button so the user can give us an object rather than strings, this will make the API more user friendly and easier to construct classes/styles conditionally 

https://github.com/svelting/svant/blob/master/src/components/button/Button.svelte#L102